### ISSUE TITLE
blocking bad http requests from akonadi

### ISSUE THREAD

**Issue by mdik** [owncloud/calendar#780]
Hi,

I discussed this briefly on the mailing list:

Before 4.14.4 Akonadi (KDE's contacts, calendar, messaging middleware) created malformed http requests when entering curiously formed ics files for sync with a CalDAV resource – and then would go completely out of sync if the server responded with 500 or similar. ownCloud was nice in that regard because it does not give an error but just says 200 OK and then immediately forgets about the request.

Now Akonadi's behaviour changed and it actually handles 4xx and 5xx answers by giving an error and not de-syncing. So now I would like ownCloud/sabredav to speak what it really thinks of the request, so Akonadi can handle it appropriately.

I would not be afraid to make changes to the code myself, but I'd need some guidance which parts to disect.

tl;dr how to make owncloud/sabredav give 4xx on malformed http requests (for uploading events).

Sincerely,

Malte

**Comment by sedrubal** [comment]
:+1: I'd be happy, if this could be fixed ;)

**Comment by georgehrke** [comment]
Hi @mdik, 

not sure if I can completely follow how exactly malformed the events by akonadi are.
Nevertheless: All the CalDav logic is inside https://github.com/owncloud/calendar/blob/master/lib/sabre/backend.php . It would be great if you could contribute a fix! If you have any questions about sabre/backend.php, feel free to ask me

**Comment by mdik** [comment]
In the OBS packaging for Debian it rather looks like
./3rdparty/sabre/dav/lib/Sabre/CalDAV/Backend/
but it seems to cover the same functionality.

Regarding the requests, I opened a bug report: https://bugs.kde.org/show_bug.cgi?id=342031
But as I said, Akonadi behaves different in the most recent version. So now I try to bring Akonadi and ownCloud on the same page as this would be my most prefered setup.

**Comment by mdik** [comment]
... I guess that is the things why people recommended me to try 8.1... I'll do that before asking more superficial questions...

(and thank you for the answer!)

**Comment by mdik** [comment]
Hi again.

So I think the flawed flow is rather in ./lib/object.php::addFromDAVData.

The clndr_objects *is* created, but doesn't show up because the "principals url" (would you call it so) is fckd. The end of the sqlite table entry reads
```
…
END:VCALENDAR
|1430733422.R118.ics|1430733422
```

as far as i can tell the ".R118" is too much.


The function takes $uri as argument, but it is not validated in any place (only written to the database in line 184). Maybe the check could already be done in ./lib/sabre/backend.php (which calls addFromDAVData)?

**Comment by mdik** [comment]
Ok. Something that helps for this is https://github.com/mdik/calendar/commit/caf66e673f5d98253fe833e584f3f4ed74c0bbdd

I noticed that the event is also actually added in korganizer, but just not displayed.

**Comment by mdik** [comment]
I'll close this bug as it shouldn't be ownCloud's problem in the first place and kdepim developers are aware of the issue.
